redbaron/utils.py

```diff
@@ -19,7 +19,8 @@
 def baron_type_to_redbaron_classname(baron_type):
     return "".join(map(lambda x: x.capitalize(), baron_type.split("_"))) + "Node"
 
-
+from functools import lru_cache
```

Comment:
`functools.lru_cache` is python3 only and will not work in python2. It might be easy to write a cache decorator with no max size:
```
def memoize(function):
    memo = {}

    @wraps(function)
    def wrapper(*args):
        if args in memo:
            return memo[args]
        else:
            rv = function(*args)
            memo[args] = rv
            return rv
    return wrapper
```